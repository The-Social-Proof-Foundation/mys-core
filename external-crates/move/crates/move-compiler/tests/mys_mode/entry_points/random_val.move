// invalid Random by value

module a::m {
    public entry fun no_random_val(_: mys::random::Random) {
        abort 0
    }
}

module mys::random {
    struct Random has key {
        id: mys::object::UID,
    }
}

module mys::object {
    struct UID has store {
        id: address,
    }
}
