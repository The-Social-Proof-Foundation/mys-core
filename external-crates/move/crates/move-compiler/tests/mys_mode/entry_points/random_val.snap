---
source: crates/move-compiler/tests/move_check_testsuite.rs
info:
  flavor: mys
  edition: legacy
  lint: false
---
error[Mys E02002]: invalid 'entry' function signature
  ┌─ tests/mys_mode/entry_points/random_val.move:4:36
  │
4 │     public entry fun no_random_val(_: mys::random::Random) {
  │            -----                   ^  ------------------- mys::random::Random must be passed by immutable reference, e.g. '&mys::random::Random'
  │            │                       │   
  │            │                       Invalid 'entry' parameter type for parameter '_'
  │            'no_random_val' was declared 'entry' here
